lib/pbench/server/api/resources/users_api.py

```diff
@@ -11,6 +11,57 @@
 from pbench.server.api.auth import Auth
 
 
+class PasswordRecovery(Resource):
+    def __init__(self, logger, auth, mail):
+        self.logger = logger
+        self.auth = auth
+        self.mail = mail
+
+    def post(self):
+        """
+        Post request for initiating password recovery.
+        This requires a JSON data with required user fields
+        {
+            "email": "user@domain.com"
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return:
+            Success: 201 with password reset link sent to email
+            Failure: <status_Code>,
+                    response_object = {
+                        "message": "failure message"
+                    }
+        """
+        # get the post data
+        user_data = request.get_json()
+        if not user_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(HTTPStatus.BAD_REQUEST, message="Invalid json object in request")
+
+        email = user_data.get("email")
+        if not email:
+            self.logger.warning("Missing email field")
+            abort(HTTPStatus.BAD_REQUEST, message="Missing email field")
+
+        # Check if email exists in the Db, else abort
+        try:
+            user = User.query(email=user_data.get("email"))
```

Comment:
You already fetched `user_data.get("email")` into local `email` so you should use that here instead of repeating the `get`.